index.js

```diff
@@ -18,78 +17,57 @@ module.exports = function () {
 		return Promise.resolve(fullname);
 	}
 
-	return pify(npmconf.load, Promise)().then(function (conf) {
+	return pify(npmconf.load)().then(conf => {
 		fullname = conf.get('init.author.name');
 
 		if (!fullname) {
-			return fallback();
+			return fallback().then(() => fullname);
 		}
 
 		return fullname;
-	}).catch(fallback).catch(function () {});
+	}).catch(fallback).then(() => fullname).catch(() => {});
 };
 
 function fallback() {
 	if (process.platform === 'darwin') {
-		return exec('id -P')
-			.then(function (stdout) {
-				fullname = stdout.trim().split(':')[7];
-
-				if (!fullname) {
-					throw new Error();
-				}
-
-				return fullname;
-			})
-			.catch(function () {
-				// `id -P` should never fail as far as I know, but just in case:
-				return exec('osascript -e "long user name of (system info)"')
-					.then(function (stdout) {
-						fullname = stdout.trim();
-
-						return fullname;
-					});
+		return passwdUser(process.getuid())
+			.then(user => fullname = user.fullname)
+			.catch(() => {
+				return execa('osascript', ['-e', '"long user name of (system info)"'])
+					.then(res => fullname = res.stdout);
```

Comment:
Weird, I thought XO failed on this because the brackets and return are moot.

```js
.catch(() => execa('osascript', ['-e', '"long user name of (system info)"']).then(res => fullname = res.stdout);
```

Same as other blocks below.